nixos/modules/security/acme.nix

```diff
@@ -204,27 +229,48 @@ in
   config = mkMerge [
     (mkIf (cfg.certs != { }) {
 
+      assertions = let
+        certs = (mapAttrsToList (k: v: v) cfg.certs);
+      in [
+        {
+          assertion = all (certOpts: certOpts.dnsProvider == null || certOpts.webroot == null) certs;
+          message = ''
+            Options `security.acme.certs.<name>.dnsProvider` and
+            `security.acme.certs.<name>.webroot` are mutually exclusive.
+          '';
+        }
+        {
+          assertion = cfg.email != null || all (certOpts: certOpts.email != null) certs;
+          message = ''
+            You must define `security.acme.certs.<name>.email` or
```

Comment:
Is `email` really necessary? I thought it was optional. Only has to be set if you want expiration emails?

Reply:
Just gave it a shot, lego (3.2.0) produced this error:
```
2020/02/03 21:34:22 You have to pass an account (email address) to the program using --email or -m
```